openedx/core/djangoapps/bookmarks/views.py

```diff
@@ -104,23 +104,18 @@ class BookmarksListView(ListCreateAPIView, BookmarksViewMixin):
     permission_classes = (permissions.IsAuthenticated,)
     serializer_class = BookmarkSerializer
 
-    @swagger_auto_schema(
-        manual_parameters=[
-            openapi.Parameter(
-                'course_id',
-                openapi.IN_QUERY,
-                type=openapi.TYPE_STRING,
-                description="The id of the course to limit the list",
-            ),
-            openapi.Parameter(
-                'fields',
-                openapi.IN_QUERY,
-                type=openapi.TYPE_STRING,
-                description="""
-                    The fields to return: display_name, path.
-                    """,
-            ),
-        ],
+    @swagger_auto_schema
+    @api_parameter(
+        'course_id',
+        openapi.IN_QUERY,
+        type=openapi.TYPE_STRING,
+        description="The id of the course to limit the list",
+    )
+    @api_parameter(
+        'fields',
+        openapi.IN_QUERY,
+        type=openapi.TYPE_STRING,
+        description="The fields to return: display_name, path.",
```

Comment:
I thought you wanted to use inner classes for this?  I'm not sure stacked decorators are better than one large decorator.

If we are going to go down the path of stacked decorators, they should all fit a uniform naming scheme.  Maybe even something like: `@openapi.schema`, `@openapi.parameter`, etc.

Reply:
That was my original thought, but I'm not sure how that would work for views with more than one method or viewsets with actions. Would there be `GetSchema`, `PostSchema`, and `DeleteSchema` inner classes? That doesn't sound great, and nor does the amount of "magic" that our decorators would have to do to make that work.

I prefer the flat, stacked decorators over one big nested decorater, but I don't have a justification for it over person preference.

I agree that a uniforming naming scheme would be ideal.

Reply:
I find the single decorator slightly preferable to the stacked decorators. With stacked decorators we are forced to define a static function attribute (`_openapi_decorator_data`) that will be modified in different ways, depending on the order of the decorators. It's slightly harder to debug.
I don't have a strong opinion on this, though.